proposal/random-with-units.md

```diff
@@ -0,0 +1,101 @@
+# Random: Draft 1
+
+*([Issue](https://github.com/sass/sass/issues/1890))*
+
+This proposal defines the behavior of the built-in [`random()`][random]
+function.
+
+[random]: ../spec/built-in-modules/math.md#random
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+  * [Design Decisions](#design-decisions)
+* [Semantics](#semantics)
+* [Deprecation Process](#deprecation-process)
+
+## Background
+
+> This section is non-normative.
+
+Sass contains a built-in `random()` function which is [documented on the public
+site][random public], but the behavior itself is **not** [specified][].
+
+Currently the function parameter `$limit` (defaults to `null`) changes the
+behavior from returning a decimal in range `[0, 1]` to returning an integer in
+range `[1, $limit]`. An error is thrown when passing a non-numeric value.
+
+It is undefined what happens when the `$limit` parameter contains a unit (e.g.
+`5px` or `8em`), and the current behavior is unexpected as `random()` will
+[drop the units and percent symbol][issue].
+For example: `random(42px) => 28` (there is no `px`).
+
+[random public]: https://sass-lang.com/documentation/modules/math#random
+[specified]: ../spec/built-in-modules/math.md#random
+[issue]: https://github.com/sass/sass/issues/1890
+
+## Summary
+
+> This section is non-normative.
+
+The built-in `random($limit: null)` function will keep its current behavior but
+it will now return units whenever a number with units or percentage is given as
+an argument.
+
+### Design Decisions
+
+This proposal keeps the existing syntax but changes the current semantics, and
+we therefore consider it a breaking change.
+
+An alternative that has also been suggested adding an optional second parameter
+to specify units, for example `random(42, 'px')` even though this can also have
```

Comment:
Nit:

```suggestion
to specify units, for example `random(42, 'px')`, even though this can also have
```

Asides like "for example" should have commas both before and after them. The commas function to indicate the scope of the aside, like parentheses but for an aside that's more closely related to the current topic.